src/python/pants/source/source_root.py

```diff
@@ -121,14 +116,26 @@ def find_by_path(self, path: str) -> Optional[SourceRoot]:
         :return: A SourceRoot instance, or None if the path is not located under a source root
                  and `unmatched == fail`.
         """
-        matched_path = self._pattern_matcher.find_root(path)
```

Comment:
This method won't support this new facility... are we planning to remove it? Or is the idea that this is a v1 only method, effectively?

Reply:
The whole class is v1-only, yes (see its docstring).